.github/workflows/ci-build.yml

```diff
@@ -16,23 +16,21 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-          
-      - uses: actions/cache@v2
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
 
       - name: Install dependencies
-        run: go get ./...
+        run: go get -t ./...
 
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v2
-        with:
-          version: v1.39
+        run: |
+          curl -sfSL -o install.sh https://raw.githubusercontent.com/golangci/golangci-lint/v1.42.1/install.sh
+          echo "0762a15cd7ac4ef439cdfb63bf12865b53c03543c0981b5d46fb542497238b829a33334a288c09d356ec8f797804dfed56f687cbcffcf853ff93da6da4106736 install.sh" | sha512sum -c
+          sh ./install.sh -b $(go env GOPATH)/bin v1.42.1
+          ~/go/bin/golangci-lint run
```

Comment:
Maybe extract into shell script, makes it easier to reuse or port to other CI system, if necessary some point